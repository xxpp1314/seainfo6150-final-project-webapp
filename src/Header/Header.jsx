import React from 'react';
import styles from "./Header.module.css";
import { Link } from "react-router-dom";

const Header = () => {
    return (
        <header className={styles.flex}>
            <h1 className={styles.title}>Pet Home</h1> 
            <nav className={styles.container}>
                <Link className={styles.menuitem} to="/"> Home </Link>
                <div className={styles.dropdown}>
                    <button className={styles.dropbutton}>
                        Category<i className="fa fa-caret-down"></i>
                    </button>
                    <div className={styles.item}>
                        <Link className={styles.li} to="/dogs">Dogs</Link>
                        <Link className={styles.li} to="/cats">Cats</Link>
                        <Link className={styles.li} to="/rabbits">Rabbits</Link>
                        <Link className={styles.li} to="/birds">Birds</Link>
                        <Link className={styles.li} to="/others">Other Pets</Link> 
                    </div>
                </div>
                <Link className={styles.menuitem} to="/aboutus">About Us</Link> 
                <Link className={styles.menuitem} to="/contactus">Contact Us</Link>
                <Link className={styles.menuitem} to="/feedback">Feedback</Link>
            </nav>   
      </header>
    )
}

export default Header;