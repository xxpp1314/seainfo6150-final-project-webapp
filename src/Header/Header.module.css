* {    
    margin: 0;
}

.flex{
    background-image: linear-gradient(to right,rgb(83, 15, 15), rgb(49, 12, 25) 70%);
    flex-direction: row;
    align-items: center;
}

.title {
    padding: 2%;
    text-align: center;
    font-weight: bold;
    color:  rgb(221, 193, 193);
    font-family: 'Lora', serif;
}

.dropdown {
  color: rgb(83, 15, 15);
  float: left;
  font-weight: bolder;
  text-align: center;  
  overflow: hidden;
  font-family: 'Space Grotesk', sans-serif;
}

.dropbutton{
  margin: 0;
  background-color: rgb(221, 193, 193);
  font-family: inherit;
  font-weight: inherit;
  font-size: inherit;
  text-align: inherit;
  border: none;
  color: rgb(83, 15, 15);
}

.item {
  display: none;
  position: absolute;
  z-index: 1;
}

.dropbutton:hover {
  color: rgb(201, 5, 5);
}

.dropdown:hover .item{
  background-color: rgb(221, 193, 193);
  color: white;
  display: inline-flex;
  flex-direction: column;
  padding:1%;
  text-align: left;
  list-style: none;
  flex-wrap: wrap;
}

.li:hover{
  background-color: rgb(221, 193, 193);
  color: rgb(201, 5, 5);
  display: block;
}


.li {
  text-decoration: none;
  text-decoration-line: none;
  color: rgb(83, 15, 15);
  float: none;
  padding: 4px 4px;
  display: block;
  text-align: left;
}

.container { 
    box-shadow: 0px 0px 10px 0px rgba(204, 0, 0, 0.5);    
    border: 0px solid #666;
    background-color:  rgb(221, 193, 193);
    padding: 0.5%;
    border: 1px solid #666;
    display: grid;
    grid-template-columns: 1fr;
    align-items: center;
  }
  
  @media screen and (min-width: 525px) {
    .container {
      grid-template-columns: repeat(2, 1fr);
    }
  }
  
  @media screen and (min-width: 800px) {
    .container {
      grid-template-columns: repeat(3, 1fr);
    }
  }
  
  @media screen and (min-width: 900px) {
    .container {
      grid-template-columns: repeat(4, 1fr);
    }
  }

  @media screen and (min-width: 100px) {
    .container {
      grid-template-columns: repeat(5, 1fr);
    }
  }

.menuitem {
  font-family: 'Space Grotesk', sans-serif;
  padding: 0.1%;
  color: rgb(83, 15, 15);
  font-weight: bolder;
  text-align: center;
  text-decoration-line: none;
}

.menuitem:hover{
  color: rgb(201, 5, 5);
}