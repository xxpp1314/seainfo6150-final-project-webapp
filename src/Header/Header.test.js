import React from "react";
import { render } from "@testing-library/react";
import Header from "./Header.jsx";
import { BrowserRouter } from 'react-router-dom';

it("renders the Header correctly", () => {
  const { container } = render(
   <BrowserRouter>  
    <Header/>
    </BrowserRouter> 
  );
  expect(container).toMatchSnapshot();
});