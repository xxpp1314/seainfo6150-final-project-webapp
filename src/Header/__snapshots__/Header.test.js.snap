// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Header correctly 1`] = `
<div>
  <header
    class="flex"
  >
    <h1
      class="title"
    >
      Pet Home
    </h1>
    <nav
      class="container"
    >
      <a
        class="menuitem"
        href="/"
      >
         Home 
      </a>
      <div
        class="dropdown"
      >
        <button
          class="dropbutton"
        >
          Category
          <i
            class="fa fa-caret-down"
          />
        </button>
        <div
          class="item"
        >
          <a
            class="li"
            href="/dogs"
          >
            Dogs
          </a>
          <a
            class="li"
            href="/cats"
          >
            Cats
          </a>
          <a
            class="li"
            href="/rabbits"
          >
            Rabbits
          </a>
          <a
            class="li"
            href="/birds"
          >
            Birds
          </a>
          <a
            class="li"
            href="/others"
          >
            Other Pets
          </a>
        </div>
      </div>
      <a
        class="menuitem"
        href="/aboutus"
      >
        About Us
      </a>
      <a
        class="menuitem"
        href="/contactus"
      >
        Contact Us
      </a>
      <a
        class="menuitem"
        href="/feedback"
      >
        Feedback
      </a>
    </nav>
  </header>
</div>
`;
